### PR TITLE
DAOS-623 build: Remove CACHEBUST from Dockerfiles

### PR DESCRIPTION
**Pull request by brianjmurrell**
CACHEBUST was erroneously repurposed from being an always-busting time
interval to being weekly.  That needs to be repaired for use-cases where
the cache should always be busted.

CB0 is the weekly cache bust and should have been used instead of
repurposing CACHEBUST.

Interestingly an already exiting CB0 at the top of all of our Dockerfiles
forces all Dockerfiles to completely rebuild weekly.

### CODE REVIEW

**Review by JohnMalmberg — APPROVED** [review]

**Review by ashleypittman — COMMENTED** [review]
Only issue I have here is the potential for wanting to use different values for CACHEBUST later, if we decide that we do want to be able to affect this at two different levels. If we don't remove CACHEBUST then we should rename it CB1 for the reasons stated in this PR.

**Comment by brianjmurrell** [comment]
> Only issue I have here is the potential for wanting to use different values for CACHEBUST later,

The entire point of this PR is that the pre-existing `CACHEBUST` was always meant to guarantee a cache bust.  If we need more levels of granularity less than _always_ those should be new variables.

**Comment by ashleypittman** [comment]
> > The entire point of this PR is that the pre-existing `CACHEBUST` was always meant to guarantee a cache bust. If we need more levels of granularity less than _always_ those should be new variables.
> 
> I see I may have misunderstood your comment. Are you suggesting that instead of removing `CACHEBUST` from these `Dockerfile`s, instead rename it to `CB1` in case, in the future we want to give that some non-always value to increase update frequency to something more frequent than `CB0` but less frequent than always?

I think we should consider if it's likely that we'll want that, and either remove CACHEBUST as this PR does, or rename it to CB1 as it's already there and debugged.

**Review by JohnMalmberg — APPROVED** [review]

File: utils/docker/Dockerfile.centos.7 [hunk 1]
```diff
@@ -6,10 +6,11 @@
 #
 
 # Pull base image
-FROM centos:centos7
+ARG BASE_DISTRO=centos:centos7
+FROM $BASE_DISTRO
```
[the remark this answers is not in the document]
Comment (brianjmurrell) [reply utils/docker/Dockerfile.centos.7:10]: Indeed.  I only made it to maintain side-by-side comparisons of all of the Dockerfiles.

**Pull request merged**